### PR TITLE
fix IDE0002 in XMakeElements.cs

### PR DESCRIPTION
**Pull request by kasperk81**
```
D:\a\_work\_temp\temp\msbuild\src\Shared\XMakeElements.cs(44,13): warning IDE0002: Name can be simplified [D:\a\_work\_temp\temp\msbuild\src\Build\Microsoft.Build.csproj]
D:\a\_work\_temp\temp\msbuild\src\Shared\XMakeElements.cs(45,13): warning IDE0002: Name can be simplified [D:\a\_work\_temp\temp\msbuild\src\Build\Microsoft.Build.csproj]
D:\a\_work\_temp\temp\msbuild\src\Shared\XMakeElements.cs(46,13): warning IDE0002: Name can be simplified [D:\a\_work\_temp\temp\msbuild\src\Build\Microsoft.Build.csproj]
D:\a\_work\_temp\temp\msbuild\src\Shared\XMakeElements.cs(47,13): warning IDE0002: Name can be simplified [D:\a\_work\_temp\temp\msbuild\src\Build\Microsoft.Build.csproj]
D:\a\_work\_temp\temp\msbuild\src\Shared\XMakeElements.cs(48,13): warning IDE0002: Name can be simplified [D:\a\_work\_temp\temp\msbuild\src\Build\Microsoft.Build.csproj]
D:\a\_work\_temp\temp\msbuild\src\Shared\XMakeElements.cs(49,13): warning IDE0002: Name can be simplified [D:\a\_work\_temp\temp\msbuild\src\Build\Microsoft.Build.csproj]
D:\a\_work\_temp\temp\msbuild\src\Shared\XMakeElements.cs(50,13): warning IDE0002: Name can be simplified [D:\a\_work\_temp\temp\msbuild\src\Build\Microsoft.Build.csproj]
D:\a\_work\_temp\temp\msbuild\src\Shared\XMakeElements.cs(51,13): warning IDE0002: Name can be simplified [D:\a\_work\_temp\temp\msbuild\src\Build\Microsoft.Build.csproj]
D:\a\_work\_temp\temp\msbuild\src\Shared\XMakeElements.cs(53,13): warning IDE0002: Name can be simplified [D:\a\_work\_temp\temp\msbuild\src\Build\Microsoft.Build.csproj]
D:\a\_work\_temp\temp\msbuild\src\Shared\XMakeElements.cs(54,13): warning IDE0002: Name can be simplified [D:\a\_work\_temp\temp\msbuild\src\Build\Microsoft.Build.csproj]
D:\a\_work\_temp\temp\msbuild\src\Shared\XMakeElements.cs(55,13): warning IDE0002: Name can be simplified [D:\a\_work\_temp\temp\msbuild\src\Build\Microsoft.Build.csproj]
D:\a\_work\_temp\temp\msbuild\src\Shared\XMakeElements.cs(56,13): warning IDE0002: Name can be simplified [D:\a\_work\_temp\temp\msbuild\src\Build\Microsoft.Build.csproj]
```

### CODE REVIEW

File: src/Shared/XMakeElements.cs [hunk 1]
```diff
@@ -35,25 +35,25 @@ internal static class XMakeElements
...
+            output,
+            itemGroup,
+            usingTask,
+            projectExtensions,
+            onError,
+            // import, "Import" items are used by Visual Basic projects
+            importGroup,
+            choose,
+            when,
+            otherwise
```
Comment (rainersigwald) [inline src/Shared/XMakeElements.cs:56]: nit: while we're in here:

```suggestion
            otherwise,
```

**Review by rainersigwald — APPROVED** [review]

File: src/Shared/XMakeElements.cs [hunk 2]
```diff
@@ -35,25 +35,25 @@ internal static class XMakeElements
         internal const string usingTaskBody = "Task";
         internal const string sdk = "Sdk";
 
-        internal static readonly char[] InvalidTargetNameCharacters = { '$', '@', '(', ')', '%', '*', '?', '.' };
+        internal static char[] InvalidTargetNameCharacters => [ '$', '@', '(', ')', '%', '*', '?', '.' ];
```
Comment (maridematte) [inline src/Shared/XMakeElements.cs:38]: I would prefer these two to continue being `ReadOnly`.

Comment (rainersigwald) [reply src/Shared/XMakeElements.cs:38]: Good catch @maridematte!

@kasperk81 Look at https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMLoN7LpGYZRwBs6ATgKYCGwA9gHYA2AnugMYAWdVAbQC66ACo8AlswDmAdQkAXHgFEAjgFc6rAM7oAvOgHoA5ABJjAGhNQolkwAo7xgJROApE4BUTgPxOAdMboQgDchMRomBTcfIIi4lJyijwAglRUjADu+gB8hibmVsY2To5FrkUeRd5FfkWBwWFIAL5AA. See how the arrow mechanism allocates a new array on every call?

Comment (rainersigwald) [reply src/Shared/XMakeElements.cs:38]: This isn't a ROS though, it's an array.

File: src/Shared/XMakeElements.cs [hunk 3]
```diff
@@ -35,25 +35,25 @@ internal static class XMakeElements
...
-            XMakeElements.onError,
-            // XMakeElements.import "Import" items are used by Visual Basic projects
-            XMakeElements.importGroup,
-            XMakeElements.choose,
-            XMakeElements.when,
-            XMakeElements.otherwise
-        };
+        internal static readonly HashSet<string> ReservedItemNames =
+        [
+            // project, "Project" is not reserved, because unfortunately ProjectReference items already use it as metadata name.
```
Comment (maridematte) [inline src/Shared/XMakeElements.cs:43]: Please keep the `XMakeElements` in the comments, so it is clearer what it is referring.

**Review by JanKrivanek — APPROVED** [review]
Looks good as is now.

Thank you!

File: src/Shared/XMakeElements.cs (hunk 3 above)
Comment (rainersigwald) [reply src/Shared/XMakeElements.cs:43]: No strong feelings on this either way.

**Review by rainersigwald — APPROVED** [review]

**Pull request merged**